server/datastore/mysql/migrations/tables/20230926082152_AddWindowsMDMTables.go

```diff
@@ -0,0 +1,70 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20230926082152, Down_20230926082152)
+}
+
+func Up_20230926082152(tx *sql.Tx) error {
+	_, err := tx.Exec(`
+CREATE TABLE windows_mdm_pending_commands (
```

Comment:
Is there a reason we need both a pending commands table and a commands table?

Reply:
Yes, pending commands will store the queued commands coming from UI/fleetctl.

On the other hand, commands table will store the actual command sent to the wire. 

The flow looks like this:

- Fleetctl calls into Fleet server API to send a new command to a device. 
- The API ends up receiving this data and "enqueuing" this command to the pending commands table through the MDMWindowsInsertPendingCommand function. At this point, these command request are stateless as they don't contain MessageID, SessionID, and CmdID information 
- Whenever a new management request coming from the MDM client stack (device) hits the `MDE2ManagementPath` URL (mdmMicrosoftManagementEndpoint handler), the pending cmds are retrieved and converted into SynCML responded with actual state data. These commands are stored on the commands table for tracking purposes. 
- The SyncML command stored on the commands table contains state information and represents an actual command sent to the device. 
- The device will reply to these commands at some point through the management API. The response for these command contains an error code that indicates the execution state for a given command. This error code is tracked on the commands table, the entry point to know to which command a given response belongs to is the state information contained in the MessageID, SessionID, and CmdID fields.

Reply:
One thing I forgot to mention, the `windows_mdm_commands` table can contain commands that were not part of the `windows_mdm_pending_commands` table and that were generated by the Fleet server in response to an incoming MDM command.